synapse_data_warehouse/synapse_aggregate/dynamic_tables/V2.50.0__unique_user_uploads_aggregate.sql

```diff
@@ -0,0 +1,94 @@
+USE SCHEMA {{database_name}}.synapse_aggregate; --noqa: JJ01,PRS,TMP
+
+CREATE OR REPLACE DYNAMIC TABLE UNIQUE_USER_UPLOADS
+    (
+	    GRANULARITY VARCHAR(16777216) COMMENT 'The dimension of the aggregate (e.g., YEARLY, MONTHLY, DAILY).',
+	    UNIQUE_USER_COUNT INT COMMENT 'The number of distinct unique users uploading during the aggregation period.',
+	    YEAR NUMBER(38,0) COMMENT 'Year of the aggregation period.',
+        MONTH NUMBER(38,0) COMMENT 'Month of the aggregation period.',
+        DAY NUMBER(38,0) COMMENT 'Day of the aggregation period.',
+        AGGREGATE_PERIOD_START DATE COMMENT 'The time when any change to the team was made (e.g. update of the team or a change to its members).',
+        AGGREGATE_PERIOD_STOP DATE COMMENT 'The time when any change to the team was made (e.g. update of the team or a change to its members).',
+	    SNAPSHOT_DATE DATE COMMENT 'Date the aggregation was calculated and stored in the table.',
+        IS_COMPLETE BOOLEAN COMMENT 'If true, then the aggregation period is complete.'
+    )
+    TARGET_LAG = '1 day'
+    WAREHOUSE = compute_xsmall
+    COMMENT = 'This table shows the total number of unique users uploading to Synapse across yearly, monthly, and daily aggregate periods.'
+    AS
+    WITH unique_users_rollup AS (
+
+        SELECT
+            YEAR(record_date)  AS year,
+            MONTH(record_date) AS month,
+            DAY(record_date)   AS day,
+            COUNT(DISTINCT user_id) AS unique_user_count,
+            GROUPING(day)   AS g_day,
+            GROUPING(month) AS g_month,
+            GROUPING(year)  AS g_year
+        FROM {{database_name}}.SYNAPSE.FILEUPLOAD --noqa: JJ01,PRS,TMP
+        -- exclude today’s partial data to avoid confusion in the final table
+        WHERE RECORD_DATE < (SELECT MAX(RECORD_DATE) FROM {{database_name}}.SYNAPSE.FILEUPLOAD) --noqa: JJ01,PRS,TMP
```

Comment:
Example of what the latest record would be for the query run on May 12, 2025:
<img width="1532" alt="image" src="https://github.com/user-attachments/assets/2bf4d231-a5a8-4fa5-acff-dcf76c09b945" />